aten/src/ATen/native/mps/operations/Shape.mm

```diff
@@ -639,11 +639,19 @@ void check_shape_except_dim(const Tensor &first, const Tensor &second,
 
           // Create placeholders
           MPSGraphTensor* inputMPSGraphTensors[inputs.size()];
+          MPSGraphTensor* castInputMPSGraphTensors[inputs.size()];
 
-          for(int i = 0; i < inputs.size(); i++)
+          for(int i = 0; i < inputs.size(); i++) {
             inputMPSGraphTensors[i] = mpsGraphUnrankedPlaceHolder(mpsGraph, getMPSDataType(result_type(inputs)));
+            if(getMPSDataType(result_type(inputs)) == MPSDataTypeBool)
```

Comment:
Does this code work for other types? did you test with consistency test to make sure other types are working correctly ?

Reply:
Yes, this code works for all types
Also, I think we should let the test I added remain, because it fails without my change, but testConsistency passes without my change

Reply:
Okay.